oak_derive/src/lib.rs

```diff
@@ -52,11 +52,11 @@ fn struct_impls(name: &Ident, data: &syn::DataStruct) -> TokenStream {
             .collect(),
         Fields::Unit => Vec::new(),
     };
-    let body = accessors_visit(&accessors);
+    let body = accessors_fold(&accessors);
 
     quote! {
         impl ::oak_io::handle::HandleVisit for #name {
-            fn visit<F: FnMut(&mut ::oak_io::Handle)>(&mut self, visitor: F) -> F {
+            fn fold<B>(&mut self, init: B, f: fn(B, &mut ::oak_io::Handle) -> B) -> B {
```

Comment:
This seems a bit too restrictive, it can only use fn pointers, but not Fn traits. Why not go with Fn trait instead? (BTW, no need to go all the way to FnMut, unlike iter::fold, I think we can assume that the receiver is just an immutable ref for now)

Reply:
I tried doing this with `Fn` instead, but that unfortunately does not work with recursion. Since the closure can't be copied or cloned I have to pass down a reference to the closure, which ends up generating `&&&&&...closure` types. Since fn pointers are `Copy`, that does work. I had a similar problem in the original implementation, which is why the `visit` method would return the visitor again instead, hence the weird type signature.